FirebaseDatabase/Tests/Unit/Swift/DatabaseAPITests.swift

```diff
@@ -0,0 +1,598 @@
+//
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// MARK: This file is used to evaluate the experience of using Firebase APIs in Swift.
+
+import Foundation
+
+import FirebaseCore
+import FirebaseDatabase
+
+final class DatabaseAPITests {
+  func usage() {
+    // MARK: - Database
+
+    // Retrieve Database Instance
+    _ = Database.database() as Database
+    _ = Database.database(url: "url" as String) as Database
+    if let app = FirebaseApp.app() {
+      _ = Database.database(app: app as FirebaseApp, url: "url" as String) as Database
+      _ = Database.database(app: app as FirebaseApp) as Database
```

Comment:
@paulb777 @ryanwilson @maksymmalyhin  based on the style of the other API tests, this line would look like:
```swift
_ = Database.database(app: app)
```
But with the added type casts here, I think it adds a bit more context around the API's types and would test if the param/return type of `database()` were to change.

Only downside is that it kinda muddles the usage of each API since a developers wouldn't normally do this.

Perhaps we should adopt a hybrid approach that:
- demonstrates intended usage
- does type checking buildtime to catch API changes.

Possible approach:
```swift 
let database = Database.database(app: app)  // Usage example
_ = database as Database // Type check
```
I think this would be a middle ground that'd allow us to evaluate API's intended usage while also also being a compile time type check to catch API changes

Great work @jrcrespoh for approaching it from this angle, I'd like to know what you think as well!

Reply:
Another option that does both in one line is:
```swift
let database: Database = Database.database(app: app)
```
But we'd have to keep the style script from running on it because it will replace `database` with an "_" since in most cases of these tests we don't use it again.

Reply:
I do agree that the separation of developer usage and compile-time type check would be even better and could be implemented as suggested. I would also wonder if there would be a way to further separate these beyond just lines, say on separate functions (e.g `usage()` vs `explicit()`) or even separate files? I imagine further separation would make it even more readable.

Reply:
I like the clear distinction between the usage example and the type checking example.